daemon/src/monitor.rs

```diff
@@ -15,35 +16,144 @@ const FINALITY_CONFIRMATIONS: u32 = 1;
+                            actor.monitor_commit_cet_timelock(cfd.order.id, &commit_subscription).await;
+                            actor.monitor_commit_refund_timelock(&params, cfd.order.id, &commit_subscription).await;
+                            actor.monitor_refund_finality( params.clone(),cfd.order.id).await;
+                        }
+                        CetStatus::TimelockExpired => {
+                            actor.monitor_commit_refund_timelock(&params, cfd.order.id, &commit_subscription).await;
+                            actor.monitor_refund_finality( params.clone(),cfd.order.id).await;
+                        }
+                        CetStatus::Ready(_price) => {
+                            // TODO: monitor CET finality
+
+                            actor.monitor_commit_refund_timelock(&params, cfd.order.id, &commit_subscription).await;
+                            actor.monitor_refund_finality( params.clone(),cfd.order.id).await;
+                        }
+                    }
+                }
+                CfdState::MustRefund { dlc, .. } => {
+                    // TODO: CET monitoring (?) - note: would require to add CetStatus information to MustRefund
+
+                    let params = MonitorParams::from_dlc_and_timelocks(dlc.clone(), cfd.refund_timelock_in_blocks());
+                    actor.cfds.insert(cfd.order.id, params.clone());
+
+                    let commit_subscription = actor
+                        .monitor
+                        .subscribe_to((params.commit.0, params.commit.1.script_pubkey()))
+                        .await;
+
+                    actor.monitor_commit_refund_timelock(&params, cfd.order.id, &commit_subscription).await;
+                    actor.monitor_refund_finality( params.clone(),cfd.order.id).await;
+                }
+
+                // too early to monitor
+                CfdState::OutgoingOrderRequest { .. }
+                | CfdState::IncomingOrderRequest { .. }
+                | CfdState::Accepted { .. }
+                | CfdState::ContractSetup { .. }
+
+                // final states
+                | CfdState::Rejected { .. }
+                | CfdState::Refunded { .. }
+                | CfdState::SafelyAborted { .. } => ()
+            }
         }
+
+        actor
     }
 
     async fn handle_start_monitoring(&mut self, msg: StartMonitoring) -> Result<()> {
         let StartMonitoring { id, params } = msg;
 
         self.cfds.insert(id, params.clone());
+        self.monitor_all(&params, id).await;
+
+        Ok(())
+    }
 
+    async fn monitor_all(&self, params: &MonitorParams, order_id: OrderId) {
+        self.monitor_lock_finality(params, order_id).await;
+
+        self.monitor_commit_finality_and_timelocks(params, order_id)
+            .await;
+
+        self.monitor_refund_finality(params.clone(), order_id).await;
+
+        // TODO: CET subscription => Requires information from Oracle
```

Comment:
This todo is misplaced, the CET monitoring has to be started upon message from the oracle. Better remove this.